src/actions/channels.js

```diff
@@ -736,18 +741,20 @@ export function joinChannel(userId, teamId, channelId, channelName) {
                 type: ChannelTypes.JOIN_CHANNEL_SUCCESS,
             },
         ]), getState);
-        dispatch(loadRolesIfNeeded(member.roles.split(' ')));
+        if (member) {
+            dispatch(loadRolesIfNeeded(member.roles.split(' ')));
+        }
 
         return {data: {channel, member}};
     };
 }
 
-export function deleteChannel(channelId) {
+export function deleteChannel(channelId: string): ActionFunc {
     return async (dispatch, getState) => {
         const state = getState();
         const viewArchivedChannels = state.entities.general.config.ExperimentalViewArchivedChannels === 'true';
 
-        dispatch({type: ChannelTypes.DELETE_CHANNEL_REQUEST}, getState);
+        dispatch({type: ChannelTypes.DELETE_CHANNEL_REQUEST, data: null}, getState);
```

Comment:
Not really related and non-blocking, but as an FYI we don't actually need to pass in `getState` to the dispatch. `thunk` will automatically pass in `getState` for any actions that are functions

Reply:
Great!